pandas/tests/apply/test_frame_transform.py

```diff
@@ -133,32 +133,35 @@ def func(x):
 @pytest.mark.parametrize("op", [*frame_kernels_raise, lambda x: x + 1])
 def test_transform_bad_dtype(op, frame_or_series, request):
     # GH 35964
-    if op == "rank":
-        request.node.add_marker(
-            pytest.mark.xfail(
-                raises=ValueError, reason="GH 40418: rank does not raise a TypeError"
-            )
-        )
-    elif op == "ngroup":
+    if op == "ngroup":
         request.node.add_marker(
             pytest.mark.xfail(raises=ValueError, reason="ngroup not valid for NDFrame")
         )
 
     obj = DataFrame({"A": 3 * [object]})  # DataFrame that will fail on most transforms
     obj = tm.get_obj(obj, frame_or_series)
+    if op == "rank":
+        error = ValueError
+        msg = "Transform function failed"
+    else:
+        error = TypeError
+        msg = (
```

Comment:
nitpick can you use the `msg = "|".join([...])` pattern here

Reply:
Thanks - agreed that's much easier to read.